model/src/main/java/org/projectnessie/model/Validation.java

```diff
@@ -25,11 +25,14 @@
 
 /** Collection of validation rules. */
 public final class Validation {
-  public static final String HASH_REGEX = "^[0-9a-fA-F]{8,64}$";
-  public static final String REF_NAME_REGEX =
-      "^[A-Za-z](((?![.][.])[A-Za-z0-9./_-])*[A-Za-z0-9._-])?$";
+
+  public static final String HASH_RAW_REGEX = "[0-9a-fA-F]{8,64}";
+  public static final String HASH_REGEX = "^" + HASH_RAW_REGEX + "$";
+  public static final String REF_NAME_RAW_REGEX =
+      "[A-Za-z](((?![.][.])[A-Za-z0-9./_-])*[A-Za-z0-9._-])";
```

Comment:
Single trailing dots (`.`) appear to be allowed in ref names... Will this work with the `getDiff` URL pattern? Will it not cause confusion with the `...` separator of the `from` and `to` sections?

If it works well, could you add a test case to make it explicit?

Reply:
See above, `.` no longer valid at the ref-name end.